### PR TITLE
feat: Document Range

### LINKED ISSUE #151: Document Range stdlib
I added this, so I should probably add the docs.

### PR DESCRIPTION
**Pull request by ospencer**
Closes #151

### CODE REVIEW

File: src/stdlib/range.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: Range
+---
+
+Utilities for working with ranges.
```
Comment (phated) [inline src/stdlib/range.md:5]: Should we describe a range? 
```suggestion
Utilities for working with ranges. A range represents an interval, or a set of values with a beginning and an end.
```
Stole that from ruby.

File: src/stdlib/range.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+### Range.**Range**
+
+```grain
+enum Range {
+  Inclusive(Number, Number),
+  Exclusive(Number, Number),
+}
+```
+
+Ranges can be inclusive or exclusive.
```
Comment (phated) [inline src/stdlib/range.md:24]:
```suggestion
Ranges can be inclusive or exclusive. When `Inclusive`, the end value will be included in operations. When `Exclusive`, the end value will be excluded from operations.
```

File: src/stdlib/range.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Checks if the given number is within the range.
+
+### Range.**forEach**
+
+```grain
+forEach : ((Number -> Void), Range) -> Void
+```
+
+Calls the given function with each number in the range.
```
Comment (phated) [inline src/stdlib/range.md:42]:
```suggestion
Calls the given function with each number in the range. Always increases the value by `1` in each iteration, even if non-integer values were provided in the range.
```

**Review by phated — CHANGES_REQUESTED** [review]
Couple of changes we probably want.

File: src/stdlib/range.md (hunk 3 above)
Comment (ospencer) [reply src/stdlib/range.md:42]: It also decreases by 1 if the range is backwards. I'll add that in:
```suggestion
Calls the given function with each number in the range. For increasing ranges, the value is increased by `1` in each iteration, and for decreasing ranges, the value is decreased by `1`. The value is always changed by `1`, even if non-integer values were provided in the range.
```

**Review by phated — APPROVED** [review]
Excellent! Approved 👍

**Pull request merged**